packages/babel-helper-builder-react-jsx-experimental/src/index.js

```diff
@@ -138,136 +138,134 @@ You can set \`throwIfNamespace: false\` to bypass this warning.`,
 
     Program: {
       enter(path, state) {
-        if (hasJSX(path)) {
```

Comment:
I'm assuming the reason why this was done originally was for performance? And that there aren't plugins that convert non-jsx stuff but that I guess isn't the case then

Reply:
Yeah probably for performance, but doing a deep traversal to check if there is JSX isn't going to be fast anyway.

Reply:
Makes me think this sort of thing shouldn't need to be set every file if it's the same for all of them (for the most part unless you use comments or overrides)?

Reply:
Yeah, usually you just set it once in the config file.